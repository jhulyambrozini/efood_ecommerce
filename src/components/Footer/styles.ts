import styled from "styled-components";
import { colors } from "../../styles";

export const FooterContainer = styled.footer`
  width: 100vw;
  background-color: ${colors.primaryLight};
  text-align: center;

   > img {
    width: 7.8rem;
    height: 3.6rem;
    margin-top: 2.5rem;
  }

  p {
    color: ${colors.primaryDark};
    font-size: 0.625rem;
  }

`

export const SocialMedia = styled.div`
 display: flex;
 gap: .5rem;
 justify-content: center;
 align-items: center;
 margin-top: 2rem;
 margin-bottom: 5rem;

  img {
    width: 1.5rem;
    height: 1.5rem;
 }
`
