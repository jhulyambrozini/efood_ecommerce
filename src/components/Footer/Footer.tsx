import logoImage from '../../assets/images/logo.png'
import instagramIcon from '../../assets/images/instagram-icon.svg'
import facebookIcon from '../../assets/images/facebook-icon.svg'
import twitterIcon from '../../assets/images/twitter-icon.svg'

import { FooterContainer, SocialMedia } from './styles'
import { Logo } from '../../styles'

const Footer = () => (
  <FooterContainer>
    <Logo src={logoImage} alt="EFOOD" />
    <SocialMedia>
      <img src={instagramIcon} alt="Icone do instagram" />
      <img src={facebookIcon} alt="Icone do facebook" />
      <img src={twitterIcon} alt="Icone do twitter" />
    </SocialMedia>
    <p>
      A efood é uma plataforma para divulgação de estabelecimentos responsabilidade pela entrega, qualidade <br/> dos produtos é toda do estabelecimento contratado.{' '}
    </p>
  </FooterContainer>
)

export default Footer
