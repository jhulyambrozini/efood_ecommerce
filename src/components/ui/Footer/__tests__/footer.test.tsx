import '@testing-library/jest-dom'
import { MemoryRouter } from 'react-router-dom'
import { render, screen } from '@testing-library/react'
import Footer from '..'

describe('<Footer />', () => {
  it('should render correctly', () => {
    render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    )
    expect(screen.getByAltText('EFOOD'))
  })
  it('should match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
