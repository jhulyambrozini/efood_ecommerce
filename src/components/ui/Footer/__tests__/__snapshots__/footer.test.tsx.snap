// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<footer
  class="sc-dcJsrY dKkVer"
>
  <a
    href="/"
    title="Home"
  >
    <img
      alt="EFOOD"
      class="sc-aXZVg drWPxd"
      src="test-file-stub"
    />
  </a>
  <div
    class="sc-iGgWBj lkGQIz"
  >
    <a
      href="#"
      title="Instagram"
    >
      <img
        alt="Icone do instagram"
        src="test-file-stub"
      />
    </a>
    <a
      href="#"
      title="Facebook"
    >
      <img
        alt="Icone do facebook"
        src="test-file-stub"
      />
    </a>
    <a
      href="#"
      title="Twitter"
    >
      <img
        alt="Icone do twitter"
        src="test-file-stub"
      />
    </a>
  </div>
  <p>
    A efood é uma plataforma para divulgação de estabelecimentos responsabilidade pela entrega, qualidade 
    <br />
     dos produtos é toda do estabelecimento contratado.
     
  </p>
</footer>
`;
